src/eu/eutampieri/catacombs/ui/gamefx/AssetManager.java

```diff
@@ -0,0 +1,196 @@
+import java.util.Map.Entry;
+
+import eu.eutampieri.catacombs.ui.utils.ImageRotator;
+
+public class AssetManager {
+	
+	private  HashMap<String,BufferedImage[]> allAnimations = new HashMap<String,BufferedImage[]>();
+	private  HashMap<String,BufferedImage> allImages = new HashMap<String,BufferedImage>();
+	
+	private ImageRotator imageRot;
+	
+	public BufferedImage[] getFrames(String key) {
+		return this.allAnimations.get(key);
+	}
+	
+	public BufferedImage getImage(String key) {
+		return this.allImages.get(key);
+	}
+	
+	public String getImageKey(BufferedImage image) {
+		for (Entry<String,BufferedImage> entry : this.allImages.entrySet()) {
+			if (entry.getValue() == image) {
+				return entry.getKey();
+			}
+		}
+		return null;
+	}
+	
+	public void load() {
+		loadSlimeAnimations();
+		loadBatAnimations();
+		loadPlayerAnimations();
+		loadImages();
+	}
+	
+	public BufferedImage horizontalFlip(BufferedImage image) {
+		int width = image.getWidth();
+		int height = image.getHeight();
+		BufferedImage flippedImage = new BufferedImage(width, height, image.getType());
+		Graphics2D g = flippedImage.createGraphics();
+		g.drawImage(image ,0 ,0 ,width ,height ,width ,0 ,0 ,height , null);
+		g.dispose();
+		return flippedImage;
+	}
+	
+	public void loadImages() {
+		// Tiles
+		GameSheets tileSheet = new GameSheets("/tileSheet.png");
+		int count = 1;
+		BufferedImage image = tileSheet.cutImage(112, 0, 16, 16);
+		
+		this.allImages.put("" + count++, image);
+		allImages.put("" + count++, imageRot.rotate(image, 90));
+		allImages.put("" + count++, imageRot.rotate(image, 180));
+		allImages.put("" + count++, tileSheet.cutImage(112 + 16, 0, 16, 16));
+		allImages.put("" + count++, tileSheet.cutImage(112 + 16, 0, 16, 16));
+		allImages.put("" + count++, tileSheet.cutImage(112, 16, 16, 16));
+		allImages.put("" + count++, tileSheet.cutImage(112 + 16, 16, 16, 16));
+		allImages.put("" + count++, tileSheet.cutImage(112 + 16, 16, 16, 16));
+		
+		for (int i = 0; i < 8; i++) {
+			for (int j = 0; i < 7; j++) {
+				if (i == 0 && j < 3) {
+					BufferedImage img;
+					img = tileSheet.cutImage(j * 16, i * 16, 16, 16);
+					this.allImages.put("" + count++, img);
+					this.allImages.put("" + count++, imageRot.rotate(image, 90));
+					this.allImages.put("" + count++, imageRot.rotate(image, 180));
+					this.allImages.put("" + count++, imageRot.rotate(image, 270));
+				} else {
+					this.allImages.put("" + count++, tileSheet.cutImage(j * 16, i * 16, 16, 16));
+				}
+			}
+		}
+		
+		// background
+		this.allImages.put("background", tileSheet.cutImage(64, 96, 16, 16));
+		
+		// world objects
+		allImages.put("coin", tileSheet.cutImage(0, 144, 16,16));
+		allImages.put("skull", tileSheet.cutImage(16, 144, 16,16));
+		allImages.put("fire", tileSheet.cutImage(0, 160, 8, 21));
+		allImages.put("potion", tileSheet.cutImage(0,181,9,11));
+		allImages.put("gun", tileSheet.cutImage(0, 193, 21, 17));
+	}
+	
+	public void loadPlayerAnimations() {
```

Comment:
I would add an `Animatable` interface and implement it for the necessary objects here

Reply:
fixed

Reply:
Where?